backend/src/bundles/notifications/notification.controller.ts

```diff
@@ -0,0 +1,219 @@
+import { type NotificationService } from '~/bundles/notifications/notification.service.js';
+import {
+    type ApiHandlerOptions,
+    type ApiHandlerResponse,
+    BaseController,
+} from '~/common/controller/controller.js';
+import { ApiPath } from '~/common/enums/enums.js';
+import { HttpCode, HTTPMethod } from '~/common/http/http.js';
+import { type Logger } from '~/common/logger/logger.js';
+import { idParametersValidationSchema } from '~/common/validation-schemas/validation-schemas.js';
+
+import { NotificationsApiPath } from './enums/enums.js';
+import {
+    type CreateNotificationRequestDto,
+    type NotificationGetOneRequestDto,
+    type UpdateNotificationRequestDto,
+} from './types/types.js';
+import {
+    createNotificationValidationSchema,
+    updateNotificationValidationSchema,
+} from './validation-schemas/validation-schemas.js';
+
+/**
+ * @swagger
+ * components:
+ *    schemas:
+ *      Notification:
+ *        type: object
+ *        properties:
+ *          id:
+ *            type: string
+ *            format: uuid
+ *          userId:
+ *            type: string
+ *            format: uuid
+ *          isRead:
+ *            type: boolean
+ *            default: false
+ *          type:
+ *            type: string
+ *            enum: [render]
+ */
+
+class NotificationController extends BaseController {
+    private notificationService: NotificationService;
+
+    public constructor(
+        logger: Logger,
+        notificationService: NotificationService,
+    ) {
+        super(logger, ApiPath.NOTIFICATIONS);
+
+        this.notificationService = notificationService;
+
+        this.addRoute({
+            path: NotificationsApiPath.ROOT,
+            method: HTTPMethod.GET,
+            handler: () => this.findAll(),
+        });
+
+        this.addRoute({
+            path: NotificationsApiPath.ROOT,
+            method: HTTPMethod.POST,
+            validation: {
+                body: createNotificationValidationSchema,
+            },
+            handler: (options) =>
+                this.create(
+                    options as ApiHandlerOptions<{
+                        body: CreateNotificationRequestDto;
+                    }>,
+                ),
+        });
+
+        this.addRoute({
+            path: NotificationsApiPath.ID,
+            method: HTTPMethod.PATCH,
+            validation: {
+                body: updateNotificationValidationSchema,
+                params: idParametersValidationSchema,
+            },
+            handler: (options) =>
+                this.update(
+                    options as ApiHandlerOptions<{
+                        body: UpdateNotificationRequestDto;
+                        params: NotificationGetOneRequestDto;
+                    }>,
+                ),
+        });
+    }
+
+    /**
+     * @swagger
+     * /api/v1/notifications/:
```

Comment:
I think you don't need to specify /api/v1/

Reply:
This is the complete endpoint, if we'd want to test it on there

Reply:
You should follow the style we have, all other endpoint do not include `/api/v1/` in their swagger docs

Reply:
Yeah you're right, but this is the correct endpoint. If we want to use swagger, what's the point to use it if it's not working properly to test and document API correctly ?

Reply:
Ok I will create a base `/api/v1` for all endpoints